saleor/core/permissions.py

```diff
@@ -0,0 +1,30 @@
+from django.contrib.auth.models import Permission
+from django.db.models import Q
+
+from ..settings import GROUP_PERMISSIONS_MODELS
+
+
+def get_user_groups(user):
+    return {'groups': list(user.groups.values_list('name', flat=True))}
+
+
+def get_permissions():
+    app_models_dict = {}
+    permissions = Permission.objects.all()
+
+    for app_model in GROUP_PERMISSIONS_MODELS:
+        app, model = app_model.split(".")
+        app_models_dict.setdefault(app, []).append(model)
+
+    q = Q()
+
+    for app, models in app_models_dict.items():
+        q |= Q(content_type__app_label=app, content_type__model__in=models)
+        q &= ~Q(content_type__app_label=app, codename__startswith='add_')
+        q &= ~Q(content_type__app_label=app, codename__startswith='change_')
+        q &= ~Q(content_type__app_label=app, codename__startswith='delete_')
```

Comment:
This seems like a very inefficient way to fetch data. Why not match how `has_perm` works and query for a list of known appname+codename pairs?

Reply:
This is used only to construct ModelForm when creating/editing a group. Has_perm can be used for User object.
This part was completely redesign so it is not working as in earliers commits.

Reply:
I understand how it works. The point I'm trying to make is that you could change the `GROUP_PERMISSIONS_MODELS ` setting to use the same syntax that `has_perm` uses and end up with a much simpler query that did not require costly `__startswith` lookups.

Reply:
By the way, is there a reason for `GROUP_PERMISSIONS_MODELS` to live in `settings`? I don't think it's likely to change unless you add more views.